src/resources/clickHouseSync/index.ts

```diff
@@ -0,0 +1,173 @@
+import * as k8s from '@pulumi/kubernetes';
+import {
+  ComponentResource,
+  CustomResourceOptions,
+  Input,
+} from '@pulumi/pulumi';
+
+import { urnPrefix } from '../../constants';
+import { AdhocEnv } from '../../utils';
+import {
+  Image,
+  Resources,
+  commonLabels,
+  configureResources,
+  image,
+} from '../../kubernetes';
+import { ClickHouseSyncConfig, loadConfig } from './utils';
+import { stringify } from 'yaml';
+import { createHash } from 'crypto';
+
+export type ClickHouseSyncArgs = AdhocEnv & {
+  props: {
+    path: string;
+    keys: ClickHouseSyncConfig;
+    vars?: Record<string, Input<string>>;
+  };
+  deployment?: {
+    name?: Input<string>;
+  };
+  namespace: Input<string>;
+  image?: Image;
+  resources?: Resources;
+};
+
+const defaults: {
+  image: Image;
+} = {
+  image: {
+    repository: 'altinity/clickhouse-sink-connector',
+    digest:
+      'sha256:1c0db9877331a0aaceb2e0f2838db3a5156a3cfbfdb83a92235b6287b576c5d0',
+  },
+};
+
+export class ClickHouseSync extends ComponentResource {
+  private deployment: k8s.apps.v1.Deployment;
+  private config: k8s.core.v1.Secret;
+
+  constructor(
+    name: string,
+    args: ClickHouseSyncArgs,
+    resourceOptions?: CustomResourceOptions,
+  ) {
+    super(`${urnPrefix}:ClickHouseSync`, name, args, resourceOptions);
+
+    const deploymentName = args.deployment?.name || `${name}-clickhouse-sync`;
+
+    const config = loadConfig(args.props.path, args.props.vars).apply(
+      (config) => {
+        return Buffer.from(
+          stringify({ ...config, ...args.props.keys }),
+        ).toString('base64');
+      },
+    );
+
+    const configChecksum = config.apply((configString) =>
+      createHash('sha256').update(configString).digest('hex'),
+    );
+
+    this.config = new k8s.core.v1.Secret('clickhouse-sync-config', {
```

Comment:
The resource name should be unique isn't it? Otherwise we may not create multiple clickhouse sync

Reply:
It would've limited to one per project, but I've added `name` to it to allow for more even.